test/integration/consul-container/libs/cluster/builder.go

```diff
@@ -74,29 +72,22 @@ type BuildOptions struct {
 	InjectAutoEncryption bool
 
 	// AllowHTTPAnyway ensures that the HTTP API is enabled even when
-	// InjectCerts or InjectAutoEncryption are enabled.
+	// InjectCerts or InjectAutoEncryption are enabled. When enabled, services
+	// will default to using plaintext to talk to the agent, regardless of other options.
 	AllowHTTPAnyway bool
-
-	// UseAPIWithTLS ensures that any accesses for the JSON API use the https
-	// port. By default it will not.
-	UseAPIWithTLS bool
-
-	// UseGRPCWithTLS ensures that any accesses for external gRPC use the
-	// grpc_tls port. By default it will not.
-	UseGRPCWithTLS bool
```

Comment:
I think these should be implied by the configuration passed to the cluster. We can always pass an option when starting the service to override this behavior.